### DISCUSSION TITLE
Is getServerSideProps executed in browser?

### DISCUSSION THREAD

**Discussion by jimin4** [vercel/next.js#20128] [Help] [answered]
I can see that `getServerSideProps` being run when SSR happens. However is this function supposed to run on browser side as well when you do pure client-side navigation to that page? What function should be used for client-side navigation dynamic data fetching?

**Comment by oMatej** [comment, marked as answer]
Even during client-side redirection, `getServerSideProps` is still executed on the server and returns data for your pages by fetching them from `${PAGE_NAME}.json` (you can see this request in the `Network` tab of DevTools).
